Scripts/Runtime/Core/ScriptableObjectCollection.cs

```diff
@@ -288,9 +288,10 @@ public void ClearBadItems()
         {
             for (int i = items.Count - 1; i >= 0; i--)
             {
-                if (items[i].IsNull() || items[i] == null)
+                if (items[i].IsNull() || items[i] == null || items[i].Collection.IsNull() || items[i].Collection == null)
```

Comment:
This is a little weird, since this is called from the collection, and we end up with some collection being null, would be better to try to see how this happen right?

Reply:
I know! It is weird, but this is what happened:

There's an editor script on my our project that deletes some items and collections from Assets folder. We then called the reload collections, and assumed this would fix all issues, but it only really takes care of :

- null collections being removed from registry
- new items that should be added to the collection

The collections that had a specific item that ceased to exist, still maintain a null item in the collection. The weird thing is that is still returns false on the `items[i].IsNull() || items[i] == null` check. 

While debugging, that item does still exist on the `items` list but its invalid..and has a null reference to a collection.
So this seemed like the best approach for it.

Also, we can't simply be calling the Synchronize in all collections as that would be a super slow process when having many collections

Reply:
Yeah, makes sense! 

approved!